devito/arch/archinfo.py

```diff
@@ -302,6 +303,41 @@ def homogenise_gpus(gpu_infos):
     return None
 
 
+@memoized_func
+def detect_cc():
+    CUDA_SUCCESS = 0
+
+    libnames = ('libcuda.so', 'libcuda.dylib', 'cuda.dll')
+    for libname in libnames:
+        try:
+            cuda = ctypes.CDLL(libname)
+        except OSError:
+            continue
+        else:
+            break
+    else:
+        raise OSError("could not load any of: " + ' '.join(libnames))
+
+    nGpus = ctypes.c_int()
+    cc_major = ctypes.c_int()
+    cc_minor = ctypes.c_int()
+    result = ctypes.c_int()
+    device = ctypes.c_int()
+    error_str = ctypes.c_char_p()
+
+    result = cuda.cuInit(0)
+    if result != CUDA_SUCCESS:
+        cuda.cuGetErrorString(result, ctypes.byref(error_str))
```

Comment:
toremove multiple gpus

Reply:
what is the call to `cuda.cuGetErrorString` for, if you don't actually use the return value `error_str` ?

Reply:
Mainly for debugging reasons if needed in the future. I am happy to drop. Let me know your preference.